### PR TITLE
added log1p

### LINKED ISSUE #14112: log1p

### PR DESCRIPTION
**Pull request by piyushgit011**
close #14112

### CODE REVIEW

**Review by Daniel4078 — CHANGES_REQUESTED** [review]
Hello. I just found a strange,potentially datatype related problem with your code: E   AssertionError:  the results from backend torch and ground truth framework tensorflow do not match
E    inf!=87.49822998046875 
E   
E   
E   Falsifying example: test_tensorflow_log1p(
E       dtype_and_x=(['float32'], [array(1.0000001e+38, dtype=float32)]),
E       test_flags=FrontendFunctionTestFlags(
E           num_positional_args=0,
E           with_out=False,
E           inplace=False,
E           as_variable=[False],
E           native_arrays=[False],
E           generate_frontend_arrays=False,
E       ),
E       fn_tree='ivy.functional.frontends.tensorflow.math.log1p',
E       frontend='tensorflow',
E       on_device='cpu',
E   )
in addition, I think this warning is also noteworthy:   /ivy/ivy/functional/backends/numpy/elementwise.py:416: RuntimeWarning: divide by zero encountered in log1p
    return np.log1p(x, out=out)

**Comment by piyushgit011** [comment]
how divide by zero can be encountered

**Comment by Daniel4078** [comment]
> how divide by zero can be encountered

you can ignore that warning I think as it seems to be something wrong inside the numpy backend

**Comment by Daniel4078** [comment]
> what can be datatype related problem?? as in pytorch log1p have beta and threshold parameters

I am not sure why the log1p function for torch backend(https://github.com/unifyai/ivy/blob/master/ivy/functional/backends/torch/elementwise.py#L225) returns inf when the tensorflow returns a normal number. You can try research further about what has happened and it may be not datatype related problem after all.

**Comment by piyushgit011** [comment]
https://github.com/pytorch/pytorch/pull/92114 
can this be potential error??

**Review by Daniel4078 — COMMENTED** [review]
I am not sure, maybe you should ask on the discord channel here(https://discord.com/channels/799879767196958751/1028706815610978345) about this problem

**Comment by piyushgit011** [comment]
> I am not sure, maybe you should ask on the discord channel here(https://discord.com/channels/799879767196958751/1028706815610978345) about this problem

ok thanks

**Comment by Daniel4078** [comment]
Have you get any reply from the discord about the problem with torch backend?

**Comment by piyushgit011** [comment]
Yes they said to add safety_factor

File: ivy/functional/frontends/tensorflow/raw_ops.py [hunk 1]
```diff
@@ -277,9 +277,7 @@ def Log(*, x, name="Log"):
     return ivy.log(x)
 
 
-@to_ivy_arrays_and_back
-def Log1p(*, x, name="Log1p"):
-    return ivy.log1p(x)
+Log1p = to_ivy_arrays_and_back(map_raw_ops_alias(tf_frontend.math.log1p))
```
Comment (Daniel4078) [inline ivy/functional/frontends/tensorflow/raw_ops.py:280]: just found a small error here: https://github.com/unifyai/ivy/actions/runs/4724482305/jobs/8381749112?pr=14126#step:3:123
ivy/functional/backends/torch/elementwise.py:227: in log1p
    return torch.log1p(x, out=out)
E   RuntimeError: "log1p_vml_cpu" not implemented for 'Half'
seems like half is not an valid datatype for the torch backend, so I think you can try to add a unsupported dtype tag in the torch backend log1p function like explained here (https://lets-unify.ai/docs/ivy/overview/deep_dive/data_types.html#supported-and-unsupported-data-types)

**Review by Daniel4078 — CHANGES_REQUESTED** [review]
Great work! Now there is only one small problem left, check the comment below. Thanks

File: ivy/functional/frontends/tensorflow/raw_ops.py (hunk 1 above)
Comment (piyushgit011) [reply ivy/functional/frontends/tensorflow/raw_ops.py:280]: done

Comment (piyushgit011) [reply ivy/functional/frontends/tensorflow/raw_ops.py:280]: on commit in revert tensorflow pr test2 ran successfully but this time after it is again having problem

**Review by Daniel4078 — APPROVED** [review]
I found that the problem is of how the test dails to get the correct unsupported datatype from the backend function. So I think your code is good enough now. Thanks.

**Pull request merged**